aten/src/ATen/native/cpu/RangeFactoriesKernel.cpp

```diff
@@ -16,30 +16,34 @@ static void linspace_kernel(TensorIterator& iter, Scalar scalar_start, Scalar sc
   AT_DISPATCH_ALL_TYPES_AND_C10_COMPLEX(iter.dtype(), "linspace_cpu", [&]() {
     // step should be of double type for all integral types
     using step_t = std::conditional_t<std::is_integral<scalar_t>::value, double, scalar_t>;
-    int64_t idx(0);
-    scalar_t start = scalar_start.to<scalar_t>();
-    scalar_t end = scalar_end.to<scalar_t>();
-    step_t step = static_cast<step_t>(end - start) / static_cast<step_t>(steps - 1);
+    const scalar_t start = scalar_start.to<scalar_t>();
+    const scalar_t end = scalar_end.to<scalar_t>();
+    const step_t step = static_cast<step_t>(end - start) / (steps - 1);
```

Comment:
```suggestion
    const step_t step = static_cast<step_t>(end - start) / static_cast<step_t>(steps - 1);
```

Reply:
Thanks for catching this! Interesting, I was confused by this comment and thought they have removed the cast: https://github.com/pytorch/pytorch/pull/38021#discussion_r423658548

We can deal with the cast in another PR though.